modules/apps/portal/portal-log4j-test/src/test/java/com/liferay/portal/log4j/Log4jTest.java

```diff
@@ -0,0 +1,464 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.portal.log4j;
+
+import com.liferay.petra.log4j.Log4JUtil;
+import com.liferay.petra.string.StringBundler;
+import com.liferay.petra.string.StringPool;
+import com.liferay.portal.kernel.io.unsync.UnsyncByteArrayOutputStream;
+import com.liferay.portal.kernel.log.Log;
+import com.liferay.portal.kernel.log.LogFactoryUtil;
+import com.liferay.portal.kernel.util.PropsUtil;
+import com.liferay.portal.kernel.util.StreamUtil;
+import com.liferay.portal.kernel.util.StringUtil;
+import com.liferay.portal.log.Log4jLogFactoryImpl;
+import com.liferay.portal.util.PropsImpl;
+
+import java.io.ByteArrayOutputStream;
+import java.io.DataOutputStream;
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.io.PrintStream;
+import java.io.StringWriter;
+
+import java.net.URI;
+import java.net.URL;
+
+import java.text.SimpleDateFormat;
+
+import java.util.Date;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.apache.log4j.LogManager;
+import org.apache.log4j.Logger;
+import org.apache.log4j.SimpleLayout;
+import org.apache.log4j.WriterAppender;
+
+import org.junit.AfterClass;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+/**
+ * @author Hai Yu
+ */
+public class Log4jTest {
+
+	@BeforeClass
+	public static void setUpClass() throws Exception {
+		PropsUtil.setProps(new PropsImpl());
+
+		_originalOutputStream = System.out;
+
+		_printStream = new TeePrintStream(_baos, _originalOutputStream);
+
+		System.setOut(_printStream);
+
+		Thread currentThread = Thread.currentThread();
+
+		ClassLoader classLoader = currentThread.getContextClassLoader();
+
+		URL url = classLoader.getResource(
+			"com/liferay/portal/log4j/dependencies/log4j.xml");
+
+		String urlContent = "";
+
+		try (InputStream inputStream = url.openStream()) {
+			byte[] bytes = _getBytes(inputStream);
+
+			urlContent = new String(bytes, StringPool.UTF8);
+		}
+
+		String dir = StringUtil.replace(
```

Comment:
Please don't put your temp file into `user.dir`. You should use APIs about temp file (there are such APIs in JDK classes, but you can also see if we have convenient ones).